lib/tdf3/src/crypto/index.ts

```diff
@@ -69,7 +69,7 @@ export function rsaPkcs1Sha256(
  * Generate a random hex key
  * @return New key as a hex string
  */
-export function generateKey(length?: number): string {
+export async function generateKey(length?: number): Promise<string> {
```

Comment:
Why do some of the Promisified functions `return` and some `Promise.resolve()`? Guessing because they're synchronous but just need to "look like" a promise to the api?

Reply:
It is just inconsistent style. Declaring a function `async` will auto-wrap all return values with Promise.resolve so I'll do that instead.

Reply:
ohhhhh learned something new today!